src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/WorkplacesController.g.cs

```diff
@@ -104,6 +104,29 @@ public async Task<ActionResult> DeleteRefToCountry([FromRoute] System.UInt32 key
         return NoContent();
     }
     
+    public virtual async Task<ActionResult<CountryDto>> PutToCountry(System.UInt32 key, [FromBody] CountryUpdateDto country)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+        
+        var related = (await _mediator.Send(new GetWorkplaceByIdQuery(_cultureCode, key))).Select(x => x.Country).SingleOrDefault();
```

Comment:
we could improve the efficiency of the query  by doing something like 
```suggestion
        var related = (await _mediator.Send(new GetWorkplaceByIdQuery(_cultureCode, key))).Select(x => x.Country.Id).SingleOrDefault();
```

Reply:
I'd leave it as is for this case to cover composite key scenarios. Otherwise it gets to complicated like this:

```
var related = (await _mediator.Send(new GetWorkplaceByIdQuery(_cultureCode, key)))
.Select(x => new { x.Country?.Id1, x.Country?.Id2} )
.SingleOrDefault();

if(related == null || related.Id1 == null || related.Id2 == null)
return NotFound();
```

And Country is not always nullable so we have to check when generating `Country?.Id` which brings extra complexity.